src/test/scala/k/akka/openid/OpenidRouterSpec.scala

```diff
@@ -14,6 +14,8 @@ import scala.concurrent.duration.FiniteDuration
 class OpenidRouterSpec extends WordSpecLike with ScalatestRouteTest {
   implicit val sessionDuration = new FiniteDuration(1, TimeUnit.MINUTES)
 
+  implicit def toScalaOption[A](underlying: java.util.Optional[A]):Option[A] = if (underlying.isPresent) Some(underlying.get) else None

```

Comment:
It seems this line was added because of the `getHeader` on line 45.

Is it possible to use the scala model API directly with type safety instead of converting java model to scala model?